opendevin/controller/agent_controller.py

```diff
@@ -176,9 +182,19 @@ async def on_event(self, event: Event):
             self.state.outputs = event.outputs  # type: ignore[attr-defined]
             await self.set_agent_state_to(AgentState.REJECTED)
         elif isinstance(event, Observation):
+            if (
+                self._pending_action
+                and self._pending_action.is_confirmed
+                == ActionConfirmationStatus.AWAITING_CONFIRMATION
+            ):
+                return
             if self._pending_action and self._pending_action.id == event.cause:
                 await self.add_history(self._pending_action, event)
                 self._pending_action = None
+                if self.state.agent_state == AgentState.ACTION_CONFIRMED:
+                    await self.set_agent_state_to(AgentState.RUNNING)
+                if self.state.agent_state == AgentState.ACTION_REJECTED:
```

Comment:
Thanks for this, really cool addition. ❤️ 

Just a little detail here: we have in the codebase another REJECT action, with the meaning that a delegate agent rejects a task it's been given. That one involves some 40 more occurrences of "Reject"-action, enum, etc, related to that meaning, all throughout the code.

I wonder if we can avoid the potential confusion about which is which. If we consider them different enough, then maybe we can name one DENY instead?

Or if they have enough in common, like, this is a user rejection, the other is an agent rejection, maybe we can name with prefixes like 'user' and 'agent'. It seems to me the first is more likely, but I'm not fully sure. Cc: @li-boxuan

Reply:
I like the prefix idea. How about renaming this to `AgentState.USER_REJECTED` and `AgentState.USER_CONFIRMED`?

Reply:
Added changes to [2f510dc](https://github.com/OpenDevin/OpenDevin/pull/2774/commits/2f510dc71094d2bd89bfae6d3391467afc63ace1)